server/fleet/host_certificates.go

```diff
@@ -0,0 +1,145 @@
+	IsIdentity bool    `plist:"IsIdentity"`
+}
+
+func (c *MDMAppleCertificateListItem) Parse() (*HostCertificateRecord, error) {
+	hash := sha1.Sum(c.Data) // nolint:gosec
+
+	parsed, err := x509.ParseCertificate(c.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	return &HostCertificateRecord{
+		SHA1Sum:              hash[:],
+		NotValidBefore:       parsed.NotBefore,
+		NotValidAfter:        parsed.NotAfter,
+		CertificateAuthority: parsed.IsCA,
+		// TODO: we need to define methodology for determining common name analogous to osquery,
+		// which seems to preferentially use Subject.CommonName for this value:
+		// https://github.com/osquery/osquery/blob/16bb01508eeca6d663b6d4f7e15034306be0fc3d/osquery/tables/system/posix/openssl_utils.cpp#L253
+		CommonName:   parsed.Subject.CommonName,
+		KeyAlgorithm: parsed.PublicKeyAlgorithm.String(),
+		// TODO: we need to define methodology for determining key strength analogous to osquery,
+		// which describes this value as "Key size used for RSA/DSA, or curve name":
+		// https://github.com/osquery/osquery/blob/16bb01508eeca6d663b6d4f7e15034306be0fc3d/osquery/tables/system/posix/openssl_utils.cpp#L337
+		KeyStrength: 0,
+		// TODO: we need to define methodology for determining key usage analogous to osquery, which
+		// describes this as "Certificate key usage and extended key usage":
+		// https://github.com/osquery/osquery/blob/16bb01508eeca6d663b6d4f7e15034306be0fc3d/osquery/tables/system/posix/openssl_utils.cpp#L166
+		KeyUsage:                  "",
+		Serial:                    parsed.SerialNumber.String(),
+		SigningAlgorithm:          parsed.SignatureAlgorithm.String(),
+		SubjectCommonName:         parsed.Subject.CommonName,
+		SubjectCountry:            parsed.Subject.Country[0],            // TODO: confirm methodology
+		SubjectOrganization:       parsed.Subject.Organization[0],       // TODO: confirm methodology
+		SubjectOrganizationalUnit: parsed.Subject.OrganizationalUnit[0], // TODO: confirm methodology
+		IssuerCommonName:          parsed.Issuer.CommonName,
+		IssuerCountry:             parsed.Issuer.Country[0],            // TODO: confirm methodology
+		IssuerOrganization:        parsed.Issuer.Organization[0],       // TODO: confirm methodology
+		IssuerOrganizationalUnit:  parsed.Issuer.OrganizationalUnit[0], // TODO: confirm methodology
+	}, nil
+}
+
+// MdmAppleErrorChainItem is the plist model for an error chain item.
+// https://developer.apple.com/documentation/devicemanagement/certificatelistresponse/errorchainitem
+type MDMAppleErrorChainItem struct {
+	ErrorCode            int    `plist:"ErrorCode"`
+	ErrorDomain          string `plist:"ErrorDomain"`
+	LocalizedDescription string `plist:"LocalizedDescription"`
+	USEnglishDescription string `plist:"USEnglishDescription"`
+}
+
+// b64Data is a byte slice that can be base64 encoded.
+type b64Data []byte
+
+// String returns the base64-encoded string form of b
+func (b b64Data) String() string {
```

Comment:
Not clear to me how/when we will use this? If it's unclear I'd suggest starting with just a `[]byte` and see how it will be used, e.g. if it's meant to be JSON-encoded and that's where base64 is needed, Go does that [automatically](https://pkg.go.dev/encoding/json#Marshal) on `[]byte` types.